packages/init/src/lib/setupSentry.ts

```diff
@@ -3,22 +3,38 @@ import semver from "semver";
 
 import { SENTRY_EXPRESS_DSN } from "../constants";
 import { checkIsSentryEnabled } from "./checkIsSentryEnabled";
-import { version } from "../../package.json";
+import { name as pkgName, version as pkgVersion } from "../../package.json";
 
 export function setupSentry(): void {
 	if (!checkIsSentryEnabled()) {
 		return;
 	}
 
-	const isStableVersion =
-		semver.prerelease(version) === null && semver.lte("0.1.0", version);
+	const parsedPkgVersion = semver.parse(pkgVersion);
+	if (parsedPkgVersion === null) {
+		throw new Error(
+			`Package \`${pkgName}\` has an invalid version \`${pkgVersion}\` in its manifest.`,
+		);
+	}
+
+	let environment;
+	if (parsedPkgVersion.prerelease.length === 0) {
+		environment = import.meta.env.MODE || "production";
+	} else if (
+		parsedPkgVersion.prerelease[0] === "alpha" ||
+		parsedPkgVersion.prerelease[0] === "beta"
+	) {
+		environment = parsedPkgVersion.prerelease[0];
+	} else {
+		throw new Error(
```

Comment:
Throwing errors in this function may halt `@slicemachine/init`. Since this pertains only to error tracking, should we be more tolerant and allow it to fail, or do you prefer the strict requirement?

(I'm not sure if we catch the error elsewhere.)